### ISSUE TITLE
Duplicate tests/unused action in test controller raises error when used.

### ISSUE THREAD

**Issue by leonelgalan** [rails/rails#46724]
### Steps to reproduce
Duplicate tests in https://github.com/rails/rails/blob/v7.0.4/actionpack/test/controller/new_base/render_template_test.rb#L103:L111:

```ruby
test "rendering a template with local variables" do
  get :with_locals
  assert_response "The secret is area51"
end

test "rendering a template with local variables without key" do
  get :with_locals
  assert_response "The secret is area51"
end
```

Both use the `with_locals` controller action. A `with_locals_without_key` action exists but it is unused; the two relevant actions are defined in [here](https://github.com/rails/rails/blob/v7.0.4/actionpack/test/controller/new_base/render_template_test.rb#L45:L51)

```ruby
def with_locals
  render template: "locals", locals: { secret: "area51" }
end

def with_locals_without_key
  render "locals", locals: { secret: "area51" }
end
```

#### [Executable Test Case](https://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#create-an-executable-test-case) based on the _Template for Action Pack (controllers, routing) issues_:
```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem "rails", "~> 7.0.0"
end

require "rack/test"
require "action_controller/railtie"

class TestApp < Rails::Application
  config.root = __dir__
  config.hosts << "example.org"
  config.session_store :cookie_store, key: "cookie_store_key"
  secrets.secret_key_base = "secret_key_base"

  config.logger = Logger.new($stdout)
  Rails.logger  = config.logger

  routes.draw do
    get "/original" => "test#with_locals_without_key"
    get "/modified" => "test#with_locals_without_key_modified"
  end
end

require "action_view/testing/resolvers"

class TestController < ActionController::Base
  include Rails.application.routes.url_helpers

  # From actionpack/test/controller/new_base/render_template_test.rb:7
  self.view_paths = [ActionView::FixtureResolver.new(
      "locals.html.erb"            => "The secret is <%= secret %>",
    )]

  # From actionpack/test/controller/new_base/render_template_test.rb:49-51
  def with_locals_without_key
    render "locals", locals: { secret: "area51" }
  end

  # Possible Fix
  def with_locals_without_key_modified
    render "/locals", locals: { secret: "area51" }
  end
end

require "minitest/autorun"

class BugTest < Minitest::Test
  include Rack::Test::Methods

  def test_returns_server_error
    get "/original"
    assert last_response.server_error?
  end

  def test_returns_success
    get "/modified"
    assert last_response.ok?
  end

  private
    def app
      Rails.application
    end
end
```

### Expected behavior

Changing "rendering a template with local variables without key" to use `with_locals_without_key` should work.

```diff
diff --git a/actionpack/test/controller/new_base/render_template_test.rb b/actionpack/test/controller/new_base/render_template_test.rb
index 3d71184b5f..70833ae11c 100644
--- a/actionpack/test/controller/new_base/render_template_test.rb
+++ b/actionpack/test/controller/new_base/render_template_test.rb
@@ -106,7 +106,7 @@ class TestWithoutLayout < Rack::TestCase
     end
 
     test "rendering a template with local variables without key" do
-      get :with_locals
+      get :with_locals_without_key
       assert_response "The secret is area51"
     end
````

### Actual behavior

Raises an error:

```
Missing template ./locals with {:locale=>[:en], :formats=>[:html], :variants=>[], :handlers=>[:raw, :erb, :html, :builder, :ruby]}.

Searched in:
  * "test/basic.html.erb, shared.html.erb, locals.html.erb, xml_template.xml.builder, with_raw.html.erb, with_implicit_raw.html.erb, with_implicit_raw.text.erb, test/with_json.html.erb, test/with_json.json.erb, test/final.json.erb, test/with_error.html.erb"
```

### Possible Solution
Shown on the executable test: prepending a slash seems to work, but that (without locals) appears to be covered in

* "rendering a template not in a subdirectory"
* "rendering a template not in a subdirectory with a leading slash without key"

And both work, not sure why is different here. 

**UPDATE**: I'm no longer convinced this is a solution, it should work without the leading slash as well. I'll look into it, see https://github.com/rails/rails/issues/46724#issuecomment-1355182948.

```diff
diff --git a/actionpack/test/controller/new_base/render_template_test.rb b/actionpack/test/controller/new_base/render_template_test.rb
index 3d71184b5f..f211a45860 100644
--- a/actionpack/test/controller/new_base/render_template_test.rb
+++ b/actionpack/test/controller/new_base/render_template_test.rb
@@ -47,7 +47,7 @@ def with_locals
     end
 
     def with_locals_without_key
-      render "locals", locals: { secret: "area51" }
+      render "/locals", locals: { secret: "area51" }
     end
 
     def builder_template
@@ -106,7 +106,7 @@ class TestWithoutLayout < Rack::TestCase
     end
 
     test "rendering a template with local variables without key" do
-      get :with_locals
+      get :with_locals_without_key
       assert_response "The secret is area51"
     end
```

### System configuration
**Rails version**: 7.0.4

**Ruby version**: 3.1.2

**Comment by leonelgalan** [comment]
@lukesarnacki thanks for your input. I'm no longer convinced that's the solution. Based on all tests it appears it should work with and without the "/" and adding the `:template` shouldn't change that. 

My current working theory is that the controller action should stay unmodified, the text should be fixed to use the unused controller action, and something inside render without the `:template` key has to change so it works without the slash.

I'll take sometime this weekend to explore this, and if I found a solution I'll add a PR.

**Comment by lukesarnacki** [comment]
Ah, I misunderstood how it should work then, sorry for the confusion!

**Comment by lukesarnacki** [comment]
Hi @leonelgalan, sorry for the confusion! I was sure it is supposed to work like that. I checked a couple of rails versions (`7.0.4`, `6.1.7`, `5.2.8.1`) and the behavior is always the same (it fails unless there is `/locals`). So I thought that maybe somehow `template:` works differently.

But then I think you are right as Rails guides state that with or without `template:` [it should work the same](https://guides.rubyonrails.org/layouts_and_rendering.html#rendering-an-action-s-template-from-another-controller) as far as I understand.

One doubt I have - do you see any scenario in which someone would depend on that bug?

**Comment by lukesarnacki** [comment]
@leonelgalan I looked [into the source of the rendering](https://github.com/rails/rails/blob/a289f4c127913fd36868e3e799999a5e32305248/actionview/lib/action_view/rendering.rb#L159) and I think that in the end, this is intentional. Notice that in the code I linked to, `template` key is added only when there is a slash in the string which explains why it fails. And I think that it is just that guides are not super clear on this but `template: "my/template"` is equivalent to `"my/template"` but as long as there is a slash in the template string.
